File: tests/RoslynDiff.Core.Tests/TestFixtures/CSharp/ClassChanges/ClassModified_Old.cs
// Test fixture: Class modification detection
// Expected: Class "DataProcessor" should be detected as Modified
//           (base class added, interface implemented)
namespace TestFixtures;

/// <summary>
/// A data processor class with original implementation.
/// </summary>
public class DataProcessor
{
    private readonly string _name;

    public DataProcessor(string name)
    {
        _name = name;
    }

    public string Process(string input)
    {
        return input.ToUpper();
    }
}

File: tests/RoslynDiff.Core.Tests/TestFixtures/VisualBasic/ClassChanges/ClassModified_Old.vb
' Test fixture: Class modification detection
' Expected: Class "Person" should be detected as Modified
Namespace TestFixtures
    ''' <summary>
    ''' A person class for testing class modification.
    ''' </summary>
    Public Class Person
        Private _name As String

        ''' <summary>
        ''' Creates a new Person instance.
        ''' </summary>
        Public Sub New(name As String)
            _name = name
        End Sub

        ''' <summary>
        ''' Gets the person's name.
        ''' </summary>
        Public Function GetName() As String
            Return _name
        End Function
    End Class
End Namespace
